File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/domain/follow/model/Follow.java
package io.dustin.apps.board.domain.follow.model;


import io.dustin.apps.common.model.BaseEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

@Getter
@Entity
@DynamicUpdate
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Follow extends BaseEntity {

    @Builder
    public Follow(Long id, @NotNull Long followerId, @NotNull Long followingId) {
        this.id =id;
        this.followerId = followerId;
        this.followingId = followingId;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "follower_id")
    private Long followerId;

    @Column(name = "following_id")
    private Long followingId;

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/follow/model/Follow.kt
package io.dustin.apps.board.domain.follow.model

import io.dustin.apps.common.model.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.DynamicUpdate

@Entity
@DynamicUpdate
class Follow (

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "follower_id")
    val followerId: Long,

    @Column(name = "following_id")
    val followingId: Long,


) : BaseEntity()
